src/components/back-to-top.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState } from "react";
+import { FaCaretUp } from "react-icons/fa";
+
+const BackToTop: React.FC<{}> = () => {
+  const [visible, setVisible] = useState(false);
+
+  const listener: EventListener = () => {
+    if (window.scrollY > 0) {
+      setVisible(true);
+    } else {
+      setVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", listener);
+    return (): void => {
+      window.removeEventListener("scroll", listener);
+    };
+  });
+
+  const scrollToTop = (): void => {
+    window.scrollTo({
```

Comment:
`scrollTo` not support in safari include iOS phones (are we ok with it) ?
For more info: https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollTo

Reply:
Nice catch. Let me test more.